### ISSUE TITLE
Question about performance on ScanRefer

### ISSUE THREAD

**Issue by Morgansgun** [yanmin-wu/EDA#22]
When I run the train_scanrefer.sh, the result in the log of output is extremely bad, I don't konw why.

last_ position alignment Acc0.25: Top-1: 0.00032, Top-5: 0.00294, Top-10: 0.00936
last_ position alignment Acc0.50: Top-1: 0.00000, Top-5: 0.00000, Top-10: 0.00000
last_ semantic alignment Acc0.25: Top-1: 0.00179, Top-5: 0.00947, Top-10: 0.01536
last_ semantic alignment Acc0.50: Top-1: 0.00000, Top-5: 0.00032, Top-10: 0.00032
proposal_ position alignment Acc0.25: Top-1: 0.00000, Top-5: 0.00000, Top-10: 0.00000
proposal_ position alignment Acc0.50: Top-1: 0.00000, Top-5: 0.00000, Top-10: 0.00000
proposal_ semantic alignment Acc0.25: Top-1: 0.00000, Top-5: 0.00000, Top-10: 0.00000
proposal_ semantic alignment Acc0.50: Top-1: 0.00000, Top-5: 0.00000, Top-10: 0.00000
0head_ position alignment Acc0.25: Top-1: 0.00000, Top-5: 0.00000, Top-10: 0.00000
0head_ position alignment Acc0.50: Top-1: 0.00000, Top-5: 0.00000, Top-10: 0.00000
0head_ semantic alignment Acc0.25: Top-1: 0.00000, Top-5: 0.00000, Top-10: 0.00000
0head_ semantic alignment Acc0.50: Top-1: 0.00000, Top-5: 0.00000, Top-10: 0.00000
1head_ position alignment Acc0.25: Top-1: 0.00000, Top-5: 0.00000, Top-10: 0.00000
1head_ position alignment Acc0.50: Top-1: 0.00000, Top-5: 0.00000, Top-10: 0.00000
1head_ semantic alignment Acc0.25: Top-1: 0.00000, Top-5: 0.00000, Top-10: 0.00000
1head_ semantic alignment Acc0.50: Top-1: 0.00000, Top-5: 0.00000, Top-10: 0.00000
2head_ position alignment Acc0.25: Top-1: 0.00000, Top-5: 0.00000, Top-10: 0.00000
2head_ position alignment Acc0.50: Top-1: 0.00000, Top-5: 0.00000, Top-10: 0.00000
2head_ semantic alignment Acc0.25: Top-1: 0.00000, Top-5: 0.00000, Top-10: 0.00000
2head_ semantic alignment Acc0.50: Top-1: 0.00000, Top-5: 0.00000, Top-10: 0.00000
3head_ position alignment Acc0.25: Top-1: 0.00000, Top-5: 0.00000, Top-10: 0.00000
3head_ position alignment Acc0.50: Top-1: 0.00000, Top-5: 0.00000, Top-10: 0.00000
3head_ semantic alignment Acc0.25: Top-1: 0.00000, Top-5: 0.00000, Top-10: 0.00000
3head_ semantic alignment Acc0.50: Top-1: 0.00000, Top-5: 0.00000, Top-10: 0.00000
4head_ position alignment Acc0.25: Top-1: 0.00000, Top-5: 0.00000, Top-10: 0.00000
4head_ position alignment Acc0.50: Top-1: 0.00000, Top-5: 0.00000, Top-10: 0.00000
4head_ semantic alignment Acc0.25: Top-1: 0.00000, Top-5: 0.00000, Top-10: 0.00000
4head_ semantic alignment Acc0.50: Top-1: 0.00000, Top-5: 0.00000, Top-10: 0.00000

Analysis
iou@0.25
easy 0.0
hard 0.002426491578646874
vd 0.0014448257178977786
vid 0.002266431629312516
unique 0.0
multi 0.0021016194832488566
iou@0.50
easy50 0.0
hard50 0.0
vd50 0.0
vid50 0.0
unique50 0.0
multi50 0.0

**Comment by GWxuan** [comment]
I've encountered exactly the same problem. Has it been resolved?

**Comment by GWxuan** [comment]
I know the reason now; don't use a single gpu for training when running train_scanrefer_single.sh.

**Comment by xiaoyaolovlife** [comment]
![image](https://github.com/yanmin-wu/EDA/assets/170327167/56ca039a-375b-423d-922d-ee77277cda6c)
![image](https://github.com/yanmin-wu/EDA/assets/170327167/0840944f-bb31-44b0-9be1-313a0bc0a1fc)
Hello! I also encountered the same problem, and I would like to ask if you know:
1. Why using a single GPU results in high loss values?
2. I used 2 GPUs to run the program, but the loss after epoch=10 decreased very slowly compared to the author's results. Can you help analyze the reasons?

**Comment by xiaoyaolovlife** [comment]
As training continues, the loss value does not show significant decrease
![image](https://github.com/yanmin-wu/EDA/assets/170327167/dee0703a-55bb-4c30-81f7-cafd5b568c4d)

**Comment by xiaoyaolovlife** [comment]
{'num_target': 256, 'sampling': 'kps', 'num_encoder_layers': 3, 'num_decoder_layers': 6, 'self_position_embedding': 'loc_learned', 'self_attend': True, 'query_points_obj_topk': 4, 'use_contrastive_align': True, 'use_soft_token_loss': True, 'detect_intermediate': True, 'joint_det': True, 'batch_size': 12, 'dataset': ['scanrefer'], 'test_dataset': 'scanrefer', 'data_root': '/home/user1009/project/EDA/preparation/', 'use_height': False, 'use_color': True, 'use_multiview': False, 'wo_obj_name': 'None', 'butd': False, 'butd_gt': False, 'butd_cls': False, 'augment_det': True, 'num_workers': 4, 'start_epoch': 1, 'max_epoch': 400, 'optimizer': 'adamW', 'weight_decay': 0.0005, 'lr': 0.0002, 'lr_backbone': 0.002, 'text_encoder_lr': 1e-05, 'lr_scheduler': 'step', 'lr_decay_epochs': [50, 75], 'lr_decay_rate': 0.1, 'clip_norm': 0.1, 'bn_momentum': 0.1, 'syncbn': False, 'warmup_epoch': -1, 'warmup_multiplier': 100, 'checkpoint_path': None, 'log_dir': '/home/user1009/project/EDA/preparation/output/logs/eda/scanrefer/1718353528', 'print_freq': 500, 'save_freq': 3, 'val_freq': 3, 'local_rank': 0, 'ap_iou_thresholds': [0.25, 0.5], 'rng_seed': 0, 'debug': False, 'eval': False, 'eval_train': False, 'pp_checkpoint': '/home/user1009/project/EDA/preparation/gf_detector_l6o256.pth', 'reduce_lr': False}


The only difference from the parameters given by the author is in 'lr_decay_epochs': [50, 75].
But I don't think this is the reason for the worsened results.

**Comment by yanmin-wu** [comment]
We use the DDP parallel training mode, with our learning rate set by default for training on 4 GPUs. If using a single GPU or 2 GPUs, it may be necessary to appropriately lower the learning rate.
refer: https://github.com/yanmin-wu/EDA/issues/5

**Comment by gml11** [comment]
> 我现在知道原因了;运行 train_scanrefer_single.sh 时，不要使用单个 GPU 进行训练。

我没有使用单个GPU训练，但感觉分还是很低，这是什么原因呢？

**Comment by gml11** [comment]
> 当我运行 train_scanrefer.sh 时，output log 中的结果非常糟糕，我不知道为什么。
> 
> last_位置对齐 Acc0.25： Top-1： 0.00032， Top-5： 0.00294， Top-10： 0.00936 last_位置对齐 Acc0.50： Top-1： 0.00000， Top-5： 0.00000， Top-10： 0.00000 last_ semantic alignment Acc0.25： Top-1： 0.00179， Top-5： 0.00947， Top-10： 0.01536 last_ semantic alignment Acc0.50： Top-1： 0.00000， Top-5： 0.00032， Top-10： 0.00032 proposal_ 位置对齐 Acc0.25： Top-1： 0.00000， Top-5： 0.00000， Top-10： 0.00000proposal_位置对齐 Acc0.50： Top-1： 0.00000， Top-5： 0.00000， Top-10： 0.00000 proposal_ semantic alignment Acc0.25： Top-1： 0.00000， Top-5： 0.00000， Top-10： 0.00000 proposal_ semantic alignment Acc0.50： Top-1： 0.00000， Top-5： 0.00000， Top-10： 0.00000 0head_ 位置对齐 Acc0.25： Top-1： 0.00000， Top-5： 0.00000， Top-10： 0.00000 0head_ 位置对齐 Acc0.50： Top-1： 0.00000， Top-5： 0.00000， Top-10： 0.000000head_语义对齐 Acc0.25： Top-1： 0.00000， Top-5： 0.00000， Top-10： 0.00000 0head_ semantic alignment Acc0.50： Top-1： 0.00000， Top-5： 0.00000， Top-10： 0.00000 1head_ position alignment Acc0.25： Top-1： 0.00000， Top-5： 0.00000， Top-10： 0.00000 1head_ position alignment Acc0.50： Top-1： 0.00000， Top-5： 0.00000， Top-10： 0.00000 1head_ semantic alignment Acc0.25： Top-1： 0.00000， Top-5： 0.00000， Top-10： 0.000001head_ 语义对齐 Acc0.50： Top-1： 0.00000， Top-5： 0.00000， Top-10： 0.00000 2head_ 位置对齐 Acc0.25： Top-1： 0.00000， Top-5： 0.00000， Top-10： 0.00000 2head_ 位置对齐 Acc0.50： Top-1： 0.00000， Top-5： 0.00000， Top-10： 0.00000 2head_ semantic alignment Acc0.25： Top-1： 0.00000， Top-5： 0.00000， Top-10： 0.00000 2head_ semantic alignment Acc0.50： Top-1： 0.00000， Top-5： 0.00000， Top-10： 0.000003head_位置对齐 Acc0.25： Top-1： 0.00000， Top-5： 0.00000， Top-10： 0.00000 3head_ 位置对齐 Acc0.50： Top-1： 0.00000， Top-5： 0.00000， Top-10： 0.00000 3head_ semantic alignment Acc0.25： Top-1： 0.00000， Top-5： 0.00000， Top-10： 0.00000 3head_ semantic alignment Acc0.50： Top-1： 0.00000， Top-5： 0.00000， Top-10： 0.00000 4head_ 位置对齐 Acc0.25： Top-1： 0.00000， Top-5： 0.00000， Top-10： 0.000004head_位置对齐 Acc0.50： Top-1： 0.00000， Top-5： 0.00000， Top-10： 0.00000 4head_ semantic alignment Acc0.25： Top-1： 0.00000， Top-5： 0.00000， Top-10： 0.00000 4head_ semantic alignment Acc0.50： Top-1： 0.00000， Top-5： 0.00000， Top-10： 0.00000
> 
> 分析 iou@0.25 简单 0.0 困难 0.002426491578646874 vd 0.0014448257178977786 vid 0.002266431629312516 独特 0.0 多重 0.0021016194832488566 iou@0.50 easy50 0.0 困难50 0.0 vd50 0.0 vid50 0.0 独特50 0.0 多重50 0.0

请问一下您现在解决这个问题了吗？
